File: src/main/kotlin/com/stannis/dataModel/statementTypes/FunctionDefinition.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.DeclarationParent
import com.stannis.dataModel.DeclarationSpecifierParent
import com.stannis.dataModel.Statement

data class FunctionDefinition(
    override val `$type`: String? = "FunctionDefinition",
    var declaratorSpecifier: DeclarationSpecifierParent?,
    var declarator: ArrayList<FunctionDeclarator>?,
    var body: ArrayList<Statement>?,
    var cyclomaticComplexity: Int,
    var modifier: String,
    var namespace: String?
) : Statement, DeclarationParent {

    fun setCyclomatic(number: Int) {
        cyclomaticComplexity = number
    }
    fun addDeclarator(statement: Statement) {
        if (declarator == null) {
            declarator = ArrayList()
        }
        declarator!!.add(statement as FunctionDeclarator)
    }
    fun addToBody(statement: Statement) {
        if (body == null) {
            body = ArrayList()
        }
        body!!.add(statement)
    }
}

File: Interpreter/Interpreter/Models/serialize/statementTypes/FunctionDefinition.cs
using System.Collections.Generic;

namespace Interpreter.Models.serialize.statementTypes
{

    public class FunctionDefinition : IStatement
    {

        public IStatement declaratorSpecifier;

        public IList<IStatement> declarator = new List<IStatement>();

        public IList<IStatement> body = new List<IStatement>();
        
        public int cyclomaticComplexity { set; get; }

    }

};